File: backend/src/main/java/com/jbconnect/healthclinix/security/controller/SecurityController.java
package com.jbconnect.healthclinix.security.controller;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.jbconnect.healthclinix.entity.user.dto.UserRequestDTO;
import com.jbconnect.healthclinix.security.facade.SecurityFacade;

@RestController
@RequestMapping("api/v1/security")
public class SecurityController {

	private final SecurityFacade securityFacade;

	public SecurityController(SecurityFacade securityFacade) {
		this.securityFacade = securityFacade;
	}

	@PostMapping("login")
	public ResponseEntity<String> login(@RequestBody UserRequestDTO userRequestDTO) {
		String token = this.securityFacade.login(userRequestDTO);
		return new ResponseEntity<>(token, HttpStatus.OK);
	}

}

File: backendkotlin/src/main/kotlin/com/jbconnect/healthclinix/healthclinix/security/controller/SecurityController.kt
package com.jbconnect.healthclinix.healthclinix.security.controller

import com.jbconnect.healthclinix.healthclinix.entity.user.dto.UserRequestDTO
import com.jbconnect.healthclinix.healthclinix.security.facade.SecurityFacade
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController


@RestController
@RequestMapping("api/v1/security")
class SecurityController(val securityFacade : SecurityFacade ) {

    @PostMapping("login")
    fun login(@RequestBody userRequestDTO: UserRequestDTO): ResponseEntity<String> {
        val token: String = this.securityFacade.login(userRequestDTO)
        return ResponseEntity(token, HttpStatus.OK)
    }
}